framework/src/auxkernels/ArrayVarReductionAux.C

```diff
@@ -44,10 +59,10 @@ ArrayVarReductionAux::computeValue()
       return _array_variable[_qp].minCoeff();
 
     case 2:
-      return _array_variable[_qp].sum();
+      return _weights.cwiseProduct(_array_variable[_qp]).sum();
```

Comment:
I think it is viable to do the same thing for the max and min. Maybe for succinctness, we can have:
```C++
const auto weighted_values = _weights.cwiseProduct(_array_variable[_qp]);
```

Then just have:
```C++
return weighted_values.sum();
```

Reply:
I don't have a strong opinion about this, the weights could be applied before the min and max operation as well. I couldn't think of a reason someone would do this. But there might be some purpose so I could make this change.

Reply:
But for consistency at least, it probably makes sense to use the weights for all operations

Reply:
Actually can I just thrown an error if `weights` is set and the operation is min/max? If the weights are used for min/max, I'm not sure if the expectation should be that the weighted min/max value is returned or that the weights should only be used when determining the min/max values and the actual un-weighted value be returned

Reply:
You can just add an error. Since we don't have use case and there are other, less convenient ways to do a weighted min/max.